scheduler/leader.go

```diff
@@ -57,13 +62,81 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
 	// creates a listener for services
 	serviceListener := zkservice.NewServiceListener(&leader)
 
+	// kicks off the snapshot cleaning goroutine
+	go cleanSnapshots(cpClient)
+
 	// starts all of the listeners
 	zzk.Start(shutdown, conn, serviceListener, hostRegistry, snapshotListener)
 }
 
+func cleanSnapshots(cpClient dao.ControlPlane) {
+
+	cleaningInterval := time.Tick(10 * time.Minute)
+
+	// Determine the SERVICED_SNAPSHOT_TTL environemnt variable and convert to duration.
+	TTLenv, err := strconv.ParseInt(os.Getenv("SERVICED_SNAPSHOT_TTL"), 10, 64)
```

Comment:
see cli/api/daemon.go on how we handle os.GetEnv (so we can pass it in as env and cmdline arg)